resources/console/charts/backend/templates/tests/test.yaml

```diff
@@ -12,9 +12,13 @@ metadata:
 spec:
   disableConcurrency: true
   template:
+    labels:
+      app: {{ .Chart.Name }}-tests
     metadata:
       annotations:
         sidecar.istio.io/inject: "true"
+      labels:
+        app: console-{{ .Chart.Name }}-tests
```

Comment:
maybe it's not very important but why do we have two different labels here?

Reply:
Istio introduced something like `canonical service` label, which labels all the pods created with sidecars. The value of this label is taken from some heuristics:
```
The initial set of heuristics the injector uses to generate a canonical service label:
1) The service.istio.io/canonical-name label.
2) Each of the user-specified alternative service label names in order.
3) The app.kubernetes.io/name label.
4) The app label
5) The workload name (e.g. Deployment or Job name).
```
as described [here](https://docs.google.com/document/d/1EsKpYNW55-LqG08zdjp1tAd224T9IEG3CEnusmL-Joc/edit#). 

As you can see if there are no labels on the pod, the workload name is used. This causes problems in testing pods, as the octopus adds `oct-tp-testsuite-all-` before each pod name. Unfortunately, the maximum length of a label in k8s is 63 characters, what in case of testing pods is usually not sufficient. 

The mitigation is to add one of the labels mentioned above, for example `4) The app label`. As the pod names are something that might change and the problem is difficult to identify when it occurs, we added the labels to all testdefinitions.